tests/research/test_batch02_durable_evidence_retention.py

```
@@ -0,0 +1,1269 @@
+    assert "--force-with-lease" in source
+
+
+def test_isolated_git_env_strips_tls_and_injected_config(monkeypatch: pytest.MonkeyPatch):
+    import scripts.research.lib.batch02_evidence_retention as retention
+
+    monkeypatch.setenv("GIT_SSL_NO_VERIFY", "1")
+    monkeypatch.setenv("GIT_CONFIG_COUNT", "1")
+    monkeypatch.setenv("GIT_CONFIG_KEY_0", "http.sslVerify")
+    monkeypatch.setenv("GIT_CONFIG_VALUE_0", "false")
+    monkeypatch.setenv("GIT_CONFIG_PARAMETERS", "http.sslVerify=false")
+    monkeypatch.setenv("GIT_PROXY_COMMAND", "/tmp/untrusted-proxy")
```

Ruff (0.16.3)

[error] Probable insecure usage of temporary file or directory: "/tmp/untrusted-proxy"

(S108)